src/service/database/index.ts

```diff
@@ -46,9 +54,52 @@ export class DatabaseService {
 
     for (const row of data) {
       const [name, description] = row as string[];
-      databases.push(new DatabaseModel({ name, description }));
+      databases.push(new DatabaseModel({ name, description }, this.context));
     }
 
     return databases;
   }
+
+  async create(
+    name: string,
+    options: CreateDatabaseOptions = {}
+  ): Promise<DatabaseModel> {
+    if (options.fail_if_exists == undefined) {
+      options.fail_if_exists = true;
+    }
+    let query: string =
+      `CREATE DATABASE ${options.fail_if_exists ? "" : "IF NOT EXISTS "} "${name}"`;
+
+    const allParamValues = [
+      options.region, 
+      options.attached_engines, 
+      options.description
+    ];
+    const queryParameters = [];
+    if (options.region || options.attached_engines || options.description) {
+      query += " WITH ";
+      for (const [index, value] of allParamValues.entries()) {
+        if (value) {
+          query += `${this.CREATE_PARAMETER_NAMES[index]} = ?`;
+          if (
+            Array.isArray(value) &&
+            value.length > 0 &&
+            value[0] instanceof EngineModel
+          ) {
+            const engines: EngineModel[] = value as EngineModel[];
+            const newValue: string[] = engines.map(
+              (eng: EngineModel) => eng.name
+            );
+            queryParameters.push(newValue);
+          } else {
+            queryParameters.push(value);
+          }
+        }
+      }
+    }
+    await this.context.connection.execute(query, {
```

Comment:
not so sure about this code. Attached engines probably will not work. Can we use something simpler like 
```typescript
const attachedEngines = options.attachedEngines ? options.attachedEngines.map(engine => `"${engine.name}"`).join(" ") : [] 
const regionSql = options.region ? this.context.queryFormatter.format("region = ?", [options.region])  : ""
const descriptionSql = options.description ? this.context.queryFormatter.format("description = ?", [options.description]) : ""

const attachedEnginesSql = engines.length
    ? `ATTACHED_ENGINES = (${attachedEngines})`
    : "";
if (regionSql || descriptionSql || attachedEnginesSql) {
 query += "with " 
 query + = regionSql
 query+ = descriptionSql
 query+= attachedEnginesSql
}

```